Python/crossfire/crossfire/occurrences.py

```diff
@@ -0,0 +1,54 @@
+from queue import Empty, Queue
+from urllib.parse import urlencode
+
+
+class Occurrences:
+    def __init__(self, client, id_state, id_cities=None, format=None):
+        self.client = client
+        self.id_state = id_state
+        self.id_cities = id_cities
+        self.format = format
+
+        self.buffer = Queue()
+        self.next_page = 1
+
+        self.params = {
+            "idState": self.id_state,
+            "idCities": self.id_cities,
+        }
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        if self.buffer.empty():
+            if not self.next_page:
+                raise StopIteration
+            self.load_occurrences()
+
+        try:
+            occurrence = self.buffer.get_nowait()
+        except Empty:
+            raise StopIteration
+
+        return occurrence
+
+    def load_occurrences(self):
+        if not self.next_page:
+            return
+
+        self.params.update({"page": self.next_page})
+        cleaned_params = urlencode(
```

Comment:
Pq já não limpamos no `__init__`?

Reply:
Não vejo um cenário onde colocamos `params["page"] = None`. Pode me explicar quando isso aconteceria?

Reply:
> Não vejo um cenário onde colocamos `params["page"] = None`. Pode me explicar quando isso aconteceria?

[aqui](https://github.com/FelipeSBarros/crossfire/pull/31/commits/85b14bd4259bf52132dded5e690c14f5d9c7d3b4#diff-5e085d038122812a0a2087a19fad4a3e33c2c15b8519c23097c14fda5252eb6aR38)

Reply:
Mas se o valor for None nunca chegamos a executar o load_occurences, né? Que eu lembre foi exatamente por isso resolvemos setar para None, para não chamar load_occurences nesse caso…

Reply:
> Mas se o valor for None nunca chegamos a executar o load_occurences, né? Que eu lembre foi exatamente por isso resolvemos setar para None, para não chamar load_occurences nesse caso…
Tenho entendido que sim, [em `load_occurrences`](https://github.com/FelipeSBarros/crossfire/pull/31/commits/85b14bd4259bf52132dded5e690c14f5d9c7d3b4#diff-5e085d038122812a0a2087a19fad4a3e33c2c15b8519c23097c14fda5252eb6aR35-R36) a primeira coisa é confirmar se `self.next_page` é None ele j'interrompe o `load_occurrences`;
TEm algo errado?

Reply:
Errado não. Só desperdício de recurso limpar o dicionário a cada requisição se não vamos executar o `load_occurences` quando a página for `None`

Reply:
@cuducos acho que [agora]([be6c650](https://github.com/FelipeSBarros/crossfire/pull/31/commits/be6c6502b7de60cc4ddff514538e81622ac45e8e)) foi :)